usaspending_api/recipient/delta_models/recipient_lookup.py

```diff
@@ -1,30 +1,686 @@
+        OR temp_rl.zip5                        IS DISTINCT FROM rl.zip5
+        OR ARRAY_SORT(ARRAY_REMOVE(temp_rl.alternate_names, COALESCE(rl.legal_business_name, '')))
+                                         IS DISTINCT FROM rl.alternate_names
+    )
+    THEN UPDATE SET
+        rl.legal_business_name = temp_rl.legal_business_name,
+        rl.duns = temp_rl.duns,
+        rl.uei = temp_rl.uei,
+        rl.address_line_1 = temp_rl.address_line_1,
+        rl.address_line_2 = temp_rl.address_line_2,
+        rl.business_types_codes = temp_rl.business_types_codes,
+        rl.city = temp_rl.city,
+        rl.congressional_district = temp_rl.congressional_district,
+        rl.country_code = temp_rl.country_code,
+        rl.parent_duns = temp_rl.parent_duns,
+        rl.parent_legal_business_name = temp_rl.parent_legal_business_name,
+        rl.parent_uei = temp_rl.parent_uei,
+        rl.state = temp_rl.state,
+        rl.zip4 = temp_rl.zip4,
+        rl.zip5 = temp_rl.zip5,
+        rl.alternate_names = temp_rl.alternate_names,
+        rl.source = temp_rl.source,
+        rl.update_date = NOW()
+    WHEN NOT MATCHED THEN INSERT (
+        id,
+        recipient_hash,
+        legal_business_name,
+        duns,
+        uei,
+        address_line_1,
+        address_line_2,
+        business_types_codes,
+        city,
+        congressional_district,
+        country_code,
+        parent_duns,
+        parent_legal_business_name,
+        parent_uei,
+        state,
+        zip4,
+        zip5,
+        alternate_names,
+        source,
+        update_date
+    )
+    VALUES (
+        0,
+        recipient_hash,
+        legal_business_name,
+        duns,
+        uei,
+        address_line_1,
+        address_line_2,
+        business_types_codes,
+        city,
+        congressional_district,
+        country_code,
+        parent_duns,
+        parent_legal_business_name,
+        parent_uei,
+        state,
+        zip4,
+        zip5,
+        alternate_names,
+        source,
+        NOW()
+    )
+    """,
+    # -----
+    # Delete any cases of old recipients from recipient_lookup where the recipient now has a UEI
+    # -----
+    r"""
+    MERGE INTO {DESTINATION_DATABASE}.{DESTINATION_TABLE} AS rl
+    USING (
+        SELECT duns_recipient_hash
+        FROM temp.temporary_restock_recipient_lookup
+        WHERE
+            uei IS NOT NULL
+            AND duns IS NOT NULL
+    ) AS temp_rl
+    ON rl.recipient_hash = temp_rl.duns_recipient_hash AND rl.uei IS NULL
+    WHEN MATCHED
+    THEN DELETE
+    """,
+    # -----
+    # Populate recipient_lookup with incremented Primary Key fields
+    # -----
+    r"""
+    CREATE OR REPLACE TEMPORARY VIEW temp_recipient_lookup_view AS (
+        SELECT
+            {AUTO_INCREMENT_MAX_ID} + ROW_NUMBER() OVER (ORDER BY recipient_hash) AS new_id,
+            rl.*
+        FROM {DESTINATION_DATABASE}.{DESTINATION_TABLE} AS rl
+        WHERE rl.id = 0
+    )
```

Comment:
Could we continue to leave the ID column off in delta, and just let the auto-incrementing ID col in the target postgres table create the PK ids? Are they meaningful to keep in delta?

Reply:
ID col was removed from delta following conversations outside of GitHub.